### PR TITLE
Discovery Docs

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This PR adds documentation about Discovery and PeeringRequestOperator components

### CODE REVIEW

File: docs/design/discovery-component.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Discovery
+## Overview
+This component's goal is to find other clusters running Liqo around us, get information needed to pair and start peering process
+
+### Features
+List of supported features
+* ClusterID creation
+  * if not already present, during component starting, it creates new ClusterID taking the UID of first master of our cluster
+* Make our cluster discoverable by other clusters
+  * this feature can be enabled and disabled runtime setting `enableAdvertisement` flag in `ClusterConfig` CR
```
Comment (fraborg) [inline docs/design/discovery-component.md:10]:
```suggestion
  * this feature can be enabled and disabled at runtime setting `enableAdvertisement` flag in `ClusterConfig` CR
```

File: docs/design/discovery-component.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  * register Liqo service on local mDNS server and answers when someone is looking for it
+* Discovery on LAN
+  * this feature can be enabled and disabled runtime setting `enableDiscovery` flag in `ClusterConfig` CR
```
Comment (fraborg) [inline docs/design/discovery-component.md:13]:
```suggestion
  * this feature can be enabled and disabled at runtime setting `enableDiscovery` flag in `ClusterConfig` CR
```

File: docs/design/discovery-component.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  * throw mDNS is able to find registered Liqo services on current LAN, a new `ForeignCluster` CR will be created for each cluster
```
Comment (fraborg) [inline docs/design/discovery-component.md:14]: what does "throw" mean?

File: docs/design/discovery-component.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+* Discovery on WAN
+  * when a new `SearchDomain` CR is added, an operator retrieves data from DNS server, a new `ForeignCluster` CR will be created for each cluster registered to domain provided
+* Automatically join discovered clusters
+  * this feature can be enabled and disabled runtime setting `autojoin` flag in `ClusterConfig` CR
```
Comment (fraborg) [inline docs/design/discovery-component.md:18]:
```suggestion
  * this feature can be enabled and disabled at runtime setting `autojoin` flag in `ClusterConfig` CR
```

File: docs/design/discovery-component.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  * when new `ForeignCluster` is added, peering process will automatically begin
+  * this value can be overwritten for WAN discovery setting it in `SearchDomain` CR
+* Start peering process
+  * when `join` flag become true in a specific `ForeignCluster`, peering process will start for that cluster
```
Comment (fraborg) [inline docs/design/discovery-component.md:22]:
```suggestion
  * when `join` flag becomes true in a specific `ForeignCluster`, peering process will start for that cluster
```

File: docs/design/discovery-component.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+### Limitations
+List of known limitations
+* After a `ForeignCluster` is created it will never disappear
+* There is no way to set default behavior if to connect or not
+  * as for Wi-Fi connections we may remember if automatically connect to a specific service when found
+* CA used on the remote cluster has to be the same used by Deployments to authenticate API Server
+  * no way to use URL as API server address (only raw IP is supported)
+* Local cluster does not handle remote cluster CA changes
+* It is not possible to trust or not CAs and to authenticate remote cluster
+* If API Server IP change, `ForeignCluster` will point to wrong one and will never change, leading to impossibility to contact remote cluster
```
Comment (fraborg) [inline docs/design/discovery-component.md:33]:
```suggestion
* If API Server IP changes, `ForeignCluster` will continue to point to the old one and will never change, leading to impossibility to contact remote cluster
```

File: docs/design/discovery-component.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+##### Merge Logic
+
+This logic merges discovered clusters with already existent ones. Currently, it checks if there is a cluster with the same ClusterId, if not it creates a new one.
+
+#### ForeignCluster management
+
+This block is a standard Kubernetes operator that is watching on `ForeignCluster` resources.
+
+When new one is added this component retrieves CAData from the remote cluster and stores it in a secret. That secret will be used in all next interactions with remote cluster to authenticate it.
```
Comment (fraborg) [inline docs/design/discovery-component.md:63]: this secret allows authentication of home cluster to foreign cluster or the opposite (authenticate foreign cluster for home cluster to know if he truly is what he says to be?) 
From the text I understand the second one

File: docs/design/discovery-component.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+When `join` flag become true in a `ForeignCluster`, this component creates a new `PeeringRequest` CR in the remote cluster triggering peering process.
```
Comment (fraborg) [inline docs/design/discovery-component.md:65]:
```suggestion
When `join` flag becomes true in a `ForeignCluster`, this component creates a new `PeeringRequest` CR in the remote cluster triggering peering process.
```

File: docs/design/discovery-component.md (hunk 7 above)
Comment (aleoli) [reply docs/design/discovery-component.md]: yes, it's that. I tried to better specify now

**Pull request merged**